packages/office-ui-fabric-react/src/components/TeachingBubble/examples/TeachingBubble.MultiStep.Example.tsx

```diff
@@ -1,75 +1,43 @@
 import * as React from 'react';
-
 import { DefaultButton, IButtonProps } from 'office-ui-fabric-react/lib/Button';
 import { TeachingBubble } from 'office-ui-fabric-react/lib/TeachingBubble';
+import { useBoolean } from '@uifabric/react-hooks';
 
-export interface ITeachingBubbleMultiStepExampleState {
-  isTeachingBubbleVisible?: boolean;
-}
-
-export class TeachingBubbleMultiStepExample extends React.Component<{}, ITeachingBubbleMultiStepExampleState> {
-  private _menuButtonElement: HTMLElement;
-
-  constructor(props: {}) {
-    super(props);
-
-    this._onDismiss = this._onDismiss.bind(this);
-    this._onShow = this._onShow.bind(this);
+const examplePrimaryButtonProps: IButtonProps = {
+  children: 'Next',
+};
+export const TeachingBubbleMultiStepExample: React.FunctionComponent = () => {
+  const [teachingBubbleVisible, { toggle: toggleTeachingBubbleVisible }] = useBoolean(false);
 
-    this.state = {
-      isTeachingBubbleVisible: false,
-    };
-  }
-
-  public render(): JSX.Element {
-    const { isTeachingBubbleVisible } = this.state;
-    const exampleSecondaryButtonProps: IButtonProps = {
+  const exampleSecondaryButtonProps: IButtonProps = React.useMemo(
```

Comment:
Same as before. useMemo should be used for expensive functions as suggested here https://usehooks.com/useMemo/